### PR TITLE
LSIF: Replace LSIF jobs with uploads in lsif-server

### PR DESCRIPTION
**Pull request by efritz**
This PR implements part of [RFC 83](https://docs.google.com/document/d/1xp8az0MW4Y8wSU3rlMXJfBYNHeXpnoESa29Zpb3-WaQ), the effort to transition from the redis-based LISF job queue to a postgres-based uploads queue.

This PR adds the `lsif_uploads` table and the associated ORM model.

### CODE REVIEW

File: migrations/1528395624_lsif_uploads.up.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
+-- Note: `commit` is a reserved word, so it's quoted.
+
+BEGIN;
+
+CREATE TYPE lsif_upload_state AS ENUM (
```
Comment (unknwon) [inline migrations/1528395624_lsif_uploads.up.sql]: qq: I never used this feature in Postgres, is it easy to add a new state in the future?

**Comment by unknwon** [comment]
FYI, I approved Percy because I think the UI diff is unrelated to this PR...

File: migrations/1528395624_lsif_uploads.up.sql (hunk 1 above)
Comment (efritz) [reply migrations/1528395624_lsif_uploads.up.sql]: Yes: https://www.postgresql.org/docs/9.1/sql-altertype.html

**Comment by efritz** [comment]
@unknwon I'm not sure why you were added explicitly to reviewers >_>

**Comment by unknwon** [comment]
The core services team is one of code owners 😭

**Comment by efritz** [comment]
> The core services team is one of code owners 😭

Quit hogging then db then :p

**Pull request merged**